### ISSUE TITLE
Problem with Consistence.Exceptions.ExceptionDeclaration

### ISSUE THREAD

**Issue by lookyman** [consistence/coding-standard#47]
```php
use Throwable;

class InvalidResponseException extends Exception
{
    public function __construct(Validator $validator, ?Throwable $previous = null)
    {
        // ...
    }
```

This reports `Exception is not chainable. It must have optional \Throwable as last constructor argument and has "Throwable".`

**Comment by VasekPurchart** [comment]
Is this the whole class? As in there is no namespace? (I see no PHP open tag and `Exception` is not used while `Throwable` is, so it seems pretty suspicious to me).

**Comment by lookyman** [comment]
Nope, of course it's not a whole class 😊 It's just an excerpt to show the issue. The problem seems to be [here](https://github.com/consistence/coding-standard/blob/fb48c3d51bce30877454903139b96ecbe19ec14d/Consistence/Sniffs/Exceptions/ExceptionDeclarationSniff.php#L119).

**Comment by miroslavhajek** [comment]
This problem is not fixed in https://github.com/consistence-community/coding-standard/pull/3/files - condition accepts only FCQN.

Use 

```
if (
    !in_array($lastArgument->getTypeHint(), ['\Throwable', '?\Throwable', 'Throwable', '?Throwable'], true)
    && !StringHelper::endsWith($lastArgument->getTypeHint(), 'Exception')
    && !StringHelper::endsWith($lastArgument->getTypeHint(), 'Error')
)
```

instead of line [119](https://github.com/consistence-community/coding-standard/blob/4d56d41aa0f90e3d519d527acfffda3753cbba79/Consistence/Sniffs/Exceptions/ExceptionDeclarationSniff.php#L119) where was added: `$lastArgument->getTypeHint() !== '?\Throwable'`
